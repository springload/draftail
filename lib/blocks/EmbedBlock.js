import PropTypes from 'prop-types';
import React from 'react';

import { NBSP } from '../api/constants';

import Icon from '../components/Icon';

/**
 * Editor block to preview and edit embedded content.
 */
const EmbedBlock = ({ entity, isActive, onClick }) => {
    const { url, title, providerName, authorName, thumbnail } = entity.getData();

    /* eslint-disable jsx-a11y/no-static-element-interactions */
    return (
        <div>
            <Icon name="icon-media" className="RichEditor-media-icon" />

            <div className="RichEditor-media-container" onClick={onClick}>
                <span className="RichEditor-media-preview">
                    <img src={thumbnail} alt={title} />
                </span>
            </div>

            {isActive ? (
                <div className="RichEditor-media-options">
                    <h3><strong>{title}</strong></h3>
                    <p>
                        <span>URL:{NBSP}<a href={url} target="_blank" rel="noopener noreferrer">{url}</a></span>
                        <br />
                        <span>Provider:{NBSP}</span>{providerName}
                        <br />
                        <span>Author:{NBSP}</span>{authorName}
                    </p>
                </div>
            ) : null}
        </div>
    );
};

EmbedBlock.propTypes = {
    entity: PropTypes.object.isRequired,
    isActive: PropTypes.bool.isRequired,
    onClick: PropTypes.func.isRequired,
};

export default EmbedBlock;
