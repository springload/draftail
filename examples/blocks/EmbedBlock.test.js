import React from 'react';
import { shallow } from 'enzyme';

import EmbedBlock from '../blocks/EmbedBlock';

describe('EmbedBlock', () => {
    it('renders', () => {
        expect(
            shallow(
                <EmbedBlock
                    blockProps={{
                        entityConfig: {},
                        entity: {
                            getData: () => ({
                                url: 'http://www.example.com/',
                                title: 'Test title',
                                thumbnail: 'http://www.example.com/example.png',
                            }),
                        },
                    }}
                />,
            ),
        ).toMatchSnapshot();
    });
});
