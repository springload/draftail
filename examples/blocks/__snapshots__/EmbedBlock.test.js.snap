// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedBlock renders 1`] = `
<MediaBlock
  alt="Embed: Test title"
  blockProps={
    Object {
      "entity": Object {
        "getData": [Function],
      },
      "entityConfig": Object {},
    }
  }
  src="http://www.example.com/example.png"
>
  <a
    className="EmbedBlock__link"
    href="http://www.example.com/"
    rel="noopener noreferrer"
    target="_blank"
    title="http://www.example.com/"
  >
    Test title
  </a>
  <button
    className="Tooltip__button"
  >
    Edit
  </button>
  <button
    className="Tooltip__button"
  >
    Remove
  </button>
</MediaBlock>
`;
